src/components/MultiHopTrade/hooks/useThorStreamingProgress/useThorStreamingProgress.tsx

```diff
@@ -0,0 +1,130 @@
+import type { ProgressProps } from '@chakra-ui/progress'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { usePoll } from 'hooks/usePoll/usePoll'
+const POLL_INTERVAL_MILLISECONDS = 30_000 // 30 seconds
+
+type ThornodeStreamingSwapResponse = {
+  tx_id?: string
+  interval?: number
+  quantity?: number
+  count?: number
+  last_height?: number
+  trade_target?: string
+  deposit?: string
+  in?: string
+  out?: string
+  failed_swaps?: number[]
+  failed_swap_reasons?: string[]
+}
```

Comment:
Can we make this a union with THOR `ThornodeQuoteResponseError` in case anything goes wrong with either our request generation, or on the THOR side? e.g https://paw.pt/h7dosA29

https://github.com/shapeshift/web/blob/15786601ba3c853d711b83ff268b9b4a377ab6ab/src/lib/swapper/swappers/ThorchainSwapper/types.ts#L51

Reply:
🧠 good suggestion, will do!